src/iter/for_each.rs

```diff
@@ -45,6 +45,13 @@ impl<'f, F, T> Folder<T> for ForEachConsumer<'f, F>
         self
     }
 
+    fn consume_iter<I>(self, iter: I) -> Self where I: IntoIterator<Item=T> {
+        for item in iter {
```

Comment:
Here, I guess we could do `iter.into_iter().for_each(self.op)` -- this gives the iterator the maximal chance to optimize, I believe. (I'm not sure if we would get a win, as I'd expect the optimizer to handle stripping out the calls to `full` anyway, but can't hurt.)

Reply:
It would have to be `fold`, until we bump the minimum rustc to reach `Iterator::for_each`.

Reply:
or `.all(|x| { self.op(x); true })`, I suppose.